packages/types/src/proof/rlnProof.ts

```diff
@@ -1,12 +1,34 @@
-import { SemaphoreProofRequest } from "./semaphoreProof";
+import { MerkleProof } from "@cryptkeeperzk/rln-proof";
+import { ISemaphoreProofRequest } from "./semaphoreProof";
+import { RoomI as IRLNRoom } from "discreetly-interfaces";
+import { ZkCircuit, ZkInputs } from "./zkProof";
 
-export type { RLNFullProof, VerificationKey } from "rlnjs";
+export type { MessageI as IRLNMessage, RoomI as IRLNRoom } from "discreetly-interfaces";
+export type { RLNFullProof, VerificationKey } from "@cryptkeeperzk/rln-proof";
 
-export interface RLNProofRequest extends SemaphoreProofRequest {
+export interface IRLNProofRequest extends ZkCircuit, ZkInputs {
+  identitySerialized: string;
   rlnIdentifier: string;
+  message: string;
+  messageLimit: bigint | number;
+  messageId: bigint | number;
+  epoch: bigint | string;
 }
 
+/**
+ * @deprecated interface type
+ */
 export interface IRlnGenerateArgs {
   identity: string;
-  payload: RLNProofRequest;
+  payload: IRLNProofRequest;
+}
+
+export interface IRLNProverInputs {
+  rlnIdentifier: bigint;
+  identitySecret: bigint;
+  userMessageLimit: bigint;
+  messageId: bigint;
+  merkleProof: MerkleProof;
+  x: bigint;
```

Comment:
What is "x"?

Reply:
Its the messageHash following the same RLN param namings: 
https://github.com/Rate-Limiting-Nullifier/rlnjs/blob/ad44375c4120bb9bb6aaba54953c4f6329d26544/src/circuit-wrapper.ts#L75

Reply:
Let's use `messageHash` on our side. It's not obvious that x is standing for hash.